sys/vnet.txt

```diff
@@ -0,0 +1,10 @@
+include <linux/types.h>
+include <linux/byteorder/generic.h>
+
+syz_blob {
+	length		len[packet, int64]
+	reserved	const[0, int64]
+	packet		array[int8, 0:256]
+} [packed]
+
+syz_emit_ethernet(arg ptr[in, syz_blob])
```

Comment:
Won't it be simpler if we pass len/buffer as separate arguments:
```
syz_emit_ethernet(len len[packet], packet ptr[in, array[int8, 0:256]])
```
?
Then you also don't need to declare the struct in C code and copy out arguments.
Also the syz_blob memory is not necessary addressable, so executor will crash when it reads length. If we make len/buffer separate arguments, then len will be always readable, and kernel will handle unaddressable buffer with EFAULT.